api_tests/lib/ethereum.ts

```diff
@@ -179,20 +178,20 @@ export class EthereumWallet {
     public async deploy_contract(
         data: string = "0x0",
         value: BigNumber | number | string = "0",
+        chainId: number = 17,
         gasLimit = "0x3D0900"
     ) {
         const nonce = await ethersClient.getTransactionCount(this.address());
 
         value = ethers.utils.bigNumberify(value);
 
-        const network = await ethersClient.getNetwork();
```

Comment:
Careful here, I _think_ this should stay. In the end, you are connecting to this network! :)

Reply:
Let's look at the difference of behaviour:

**Before:**
The network returned by cnd for the blockchain action (deploy contract) is ignored (not even checked). The network used to do the transaction is the one that is taken from the node we are connected to.

**After:**
The chain id returned by cnd for the blockchain action (deploy contract) is used to do the transaction.

I think the **after** behaviour makes more sense.